src/opnsense/mvc/app/controllers/OPNsense/Base/ControllerRoot.php

```diff
@@ -48,6 +49,31 @@ protected function sessionClose()
     }
 
     /**
+     * Set locale for gettext
+     */
+    public static function setLocale()
```

Comment:
if we skip the static here, we could save our locale in the controller object and add a getLocale() to request our setting back (which can then be used in getTranslator() for GetText() ).